src/main/webapp/js/projects/react/components/ProjectsSearch.js

```diff
@@ -0,0 +1,61 @@
+/**
+ * @fileoverview The highest-order component of the mentored project search
+ * page. This handles the loading of project preview data.
+ */
+import ProjectList from './ProjectList.js';
+import standardizeFetchErrors from '../../../fetch_handler.js';
+
+/**
+ * The main project search page react component.
+ * @extends React.Component
+ */
+export default class ProjectsSearch extends React.Component {
+  /**
+   * Create the projects search page react component.
+   * @param {{}} props
+   */
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      isFetching: true,
+    };
+  }
+
+  /**
+   * Loads the project preview data after React has initialized the component.
+   */
+  componentDidMount() {
+    const fetchRequest = standardizeFetchErrors(
+        fetch('/project-previews'),
+        'Failed to communicate with the server, please try again later.',
+        'Encountered a server error, please try again later.');
+    fetchRequest.then((projectPreviews) => {
+      console.log('Project Previews Received:');
+      console.log(projectPreviews);
+      this.setState({
+        isFetching: false,
+        projectPreviews,
+      });
+    }).catch((errorResponse) => {
+      alert(errorResponse.userMessage);
+      console.error(
+          `Error ${errorResponse.statusCode}: ${errorResponse.error}`);
+    });
+  }
+
+  /**
+   * Renders the list of project previews or a loading notification if the data
+   * is still being loaded.
+   * @return {React.Component}
+   */
+  render() {
```

Comment:
I think that this function should be named something like `loadProjectPreviews`. `render` is used to describe logic that actually displays images on the screen (https://en.wikipedia.org/wiki/Rendering_(computer_graphics)). It seems like this function just returns data and doesn't handle any of the graphics display logic.

Reply:
You make a good point, however [this is a function associated with the React.Component class](https://reactjs.org/docs/react-component.html#render) that is being extended here. I'm going to add the @override tag as per the [style guide](https://google.github.io/styleguide/jsguide.html#jsdoc-method-and-function-comments) to make this more clear.